### PR TITLE
Pin System.Formats.Asn

### PR DESCRIPTION
**Pull request by JanKrivanek**
Fixes https://devdiv.visualstudio.com/DevDiv/_componentGovernance/DotNet-msbuild-Trusted/alert/9977351?typeId=6797870&pipelinesTrackingFilter=0

### Context
Lets pin System.Formats.Asn, till the System.Security.Cryptography.Pkcs is available and updated

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -41,6 +41,7 @@
     <SystemTextJsonVersion>8.0.3</SystemTextJsonVersion>
     <SystemThreadingChannelsVersion>8.0.0</SystemThreadingChannelsVersion>
     <SystemThreadingTasksDataflowVersion>8.0.0</SystemThreadingTasksDataflowVersion>
+	<SystemFormatsAsnVersion>8.0.1</SystemFormatsAsnVersion>
```
Comment (rainersigwald) [inline eng/Versions.props:44]:
```suggestion
    <SystemFormatsAsnVersion>8.0.1</SystemFormatsAsnVersion>
```

**Review by rainersigwald — CHANGES_REQUESTED** [review]
I think you'll also need to add one of these sections https://github.com/dotnet/msbuild/blob/85d1d666ecdf42d3e2f851f403057349f963ea0e/eng/Version.Details.xml#L34-L39 so sourcebuild can flow the right versions.

**Comment by JanKrivanek** [comment]
I added the dependency via:

```
darc add-dependency --name "System.Formats.Asn1" --version 8.0.1 --type "product" --repo "https://dev.azure.com/dnceng/internal/_git/dotnet-runtime" --commit "bf5e279d9239bfef5bb1b8d6212f1b971c434606"
```

I'm bit blurry on following - and would appreciate info or documentation pointers (tagging @dotnet/source-build-internal to chime in here):

* How should the SHA be obtained? I suppose for runtime I just grab SHA for appropriate tag - e.g. https://github.com/dotnet/runtime/tree/v8.0.1, right?
* When adding new product dependency, should I allways make sure it exist in SBRP and if not (case here), add it via https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages?

**Comment by mthalman** [comment]
> I added the dependency via:
> 
> ```
> darc add-dependency --name "System.Formats.Asn1" --version 8.0.1 --type "product" --repo "https://dev.azure.com/dnceng/internal/_git/dotnet-runtime" --commit "bf5e279d9239bfef5bb1b8d6212f1b971c434606"
> ```
> 
> I'm bit blurry on following - and would appreciate info or documentation pointers (tagging @dotnet/source-build-internal to chime in here):
> 
> * How should the SHA be obtained? I suppose for runtime I just grab SHA for appropriate tag - e.g. https://github.com/dotnet/runtime/tree/v8.0.1, right?
> * When adding new product dependency, should I allways make sure it exist in SBRP and if not (case here), add it via https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages?

Since you've defined the dependency using darc, it's set up perfectly. And since this dependency comes from runtime, which is built before msbuild in the VMR, this prebuilt will only occur in the msbuild repo's build, not the VMR's build. For that reason, it's acceptable to simply configure this as an [allowed prebuilt](https://github.com/dotnet/msbuild/blob/main/eng/SourceBuildPrebuiltBaseline.xml) rather than defining this package in SBRP.

**Review by mthalman — APPROVED** [review]

File: eng/Packages.props [hunk 2]
```diff
@@ -31,5 +31,6 @@
     <PackageVersion Include="System.Threading.Channels" Version="$(SystemThreadingChannelsVersion)" />
     <PackageVersion Include="System.Threading.Tasks.Dataflow" Version="$(SystemThreadingTasksDataflowVersion)" />
     <PackageVersion Include="xunit.console" Version="$(XUnitVersion)" />
+    <PackageVersion Include="System.Formats.Asn1" Version="$(SystemFormatsAsn1Version)" />
```
Comment (rainersigwald) [inline eng/Packages.props:34]: nittiest nit that ever nitted: alphabetical order in all these files please?

**Review by rainersigwald — APPROVED** [review]

File: eng/Version.Details.xml [hunk 3]
```diff
@@ -65,6 +65,10 @@
       <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
       <Sha>5535e31a712343a63f5d7d796cd874e563e5ac14</Sha>
     </Dependency>
+    <Dependency Name="System.Formats.Asn1" Version="8.0.1">
+      <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
+      <Sha>bf5e279d9239bfef5bb1b8d6212f1b971c434606</Sha>
```
Comment (rainersigwald) [inline eng/Version.Details.xml:70]: > * How should the SHA be obtained? I suppose for runtime I just grab SHA for appropriate tag - e.g. https://github.com/dotnet/runtime/tree/v8.0.1, right?

This is incorrect for subtle reasons. _Packages_ that build in `dotnet/runtime` rev their versions and republish _only_ when they change meaningfully, but _the overall runtime_ revs version every time. This means that 8.0.1 of some package may have been built in _any_ 8.0.x _of runtime_. In this case, https://nuget.info/packages/System.Formats.Asn1/8.0.1 says that https://github.com/dotnet/runtime/commit/2aade6beb02ea367fd97c4070a4198802fe61c03 is the relevant commit, which is https://github.com/dotnet/runtime/releases/tag/v8.0.7, so "the first time ASN1 8 had to be serviced was the seventh overall servicing release of .NET 8".

```suggestion
      <Sha>2aade6beb02ea367fd97c4070a4198802fe61c03</Sha>
```

Comment (JanKrivanek) [reply eng/Version.Details.xml:70]: This is super helpful info! Thanks!

**Pull request merged**